Fusion LSP in Insights (#7822)
fishtown-analytics/docs.getdbt.com: website/docs/docs/explore/navigate-dbt-insights.md

<Lightbox src="/img/docs/dbt-insights/insights-copilot.png" width="60%" title="dbt Insights dbt Copilot" />
<Lightbox src="/img/docs/dbt-insights/manage-bookmarks.png" width="60%" title="Manage your query bookmarks" />
</DocCarousel>

## LSP features in dbt Insights

> <!-- Added this new subsection for now, please check if the features listed here are correct. Might need screenshots-->

Author: @dave-connors-3 @aricallen Is it possible to get screenshots for the features listed here?

The following Language Server Protocol (LSP) features are available for projects upgraded to <Constant name="fusion" />:

- **Live error detection:** Automatically validate your SQL code to detect errors and surface warnings, without hitting the warehouse. This includes both dbt errors (like invalid `ref`) and SQL errors (like invalid column name or SQL syntax).
- **`Ref` suggestions:** Autocomplete model names when using the `ref()` function to reference other models in your project.
- **Hover insights:** See context on tables, columns, and functions without leaving your code. Simply hover over any SQL element to see details like column names and data types.
- **Live CTE previews:** Preview a CTE’s output directly from inside your dbt model for faster validation and debugging.

> - **Default VS Code commands:** You can use text editing capabilities such as multi-cursor editing, text transformations, and keyboard shortcuts within the <Constant name="query_page" /> interface.

Reviewer: this is not exactly an LSP feature, so we can likely exclude this from the list. This is already true in insights today!

Author: Removed here: https://github.com/dbt-labs/docs.getdbt.com/pull/7822/commits/3e7fd6a30a7eb76ce8b8fdd77c862bff785a73dd

<Lightbox src="/img/docs/dbt-insights/insights-copilot.png" width="60%" title="dbt Insights dbt Copilot" />
<Lightbox src="/img/docs/dbt-insights/manage-bookmarks.png" width="60%" title="Manage your query bookmarks" />
</DocCarousel>

> ## LSP features in dbt Insights

Reviewer 2: Since we're already on the insights page we can trim this and make it a shorter anchor link
```suggestion
## LSP features
```

The following Language Server Protocol (LSP) features are available for projects upgraded to <Constant name="fusion" />:

- **Live CTE previews:** Preview a CTE’s output for faster validation and debugging.

    <Lightbox src="/img/docs/dbt-insights/preview-cte.png" width="90%" title="Preview CTE in Insights" />

> - **Live error detection:** Automatically validate your SQL code to detect errors and surface warnings, without hitting the warehouse. This includes both dbt errors (like invalid `ref`) and SQL errors (like invalid column name or SQL syntax).

Reviewer 2: This is potentially a more marketable term than live
```suggestion
- **Real-time error detection:** Automatically validate your SQL code to detect errors and surface warnings, without hitting the warehouse. This includes both dbt errors (like invalid `ref`) and SQL errors (like invalid column name or SQL syntax).
```

    <Lightbox src="/img/docs/dbt-insights/sql-validation.png" width="90%" title="Live error detection" />

- **`ref` suggestions:** Autocomplete model names when using the `ref()` function to reference other models in your project.
  
    <Lightbox src="/img/docs/dbt-insights/ref-autocomplete.png" width="90%" title="ref suggestions in Insights" />

> - **Hover insights:** See context on tables, columns, and functions without leaving your code. Simply hover over any SQL element to see details like column names and data types.

Reviewer 2: ```suggestion
- **Hover insights:** View context on tables, columns, and functions without leaving your code. Hover over any SQL element to see details like column names and data types.
```
